code42_on_poll_connector.py

```diff
@@ -0,0 +1,287 @@
+import dateutil.parser
+import phantom.app as phantom
+from py42.sdk.queries.fileevents.file_event_query import FileEventQuery
+from py42.sdk.queries.fileevents.filters import (
+    ExposureType,
+    DeviceUsername,
+    Actor,
+    EventTimestamp,
+    EventType,
+    FileCategory,
+)
+
+from code42_util import get_thirty_days_ago, build_alerts_query
+
+"""The contents of this module that related to mapping alert observations to file events borrows heavily from the
+Code42 Cortex XSOAR integration.
+"""
+
+
+def get_file_category_value(key):
+    # Meant to handle all possible cases
+    key = key.lower().replace("-", "").replace("_", "")
+    category_map = {
+        "sourcecode": FileCategory.SOURCE_CODE,
+        "audio": FileCategory.AUDIO,
+        "executable": FileCategory.EXECUTABLE,
+        "document": FileCategory.DOCUMENT,
+        "image": FileCategory.IMAGE,
+        "pdf": FileCategory.PDF,
+        "presentation": FileCategory.PRESENTATION,
+        "script": FileCategory.SCRIPT,
+        "spreadsheet": FileCategory.SPREADSHEET,
+        "video": FileCategory.VIDEO,
+        "virtualdiskimage": FileCategory.VIRTUAL_DISK_IMAGE,
+        "archive": FileCategory.ZIP,
+    }
+    return category_map.get(key, "UNCATEGORIZED")
+
+
+class Code42OnPollConnector:
+    def __init__(self, connector, client, state):
+        super(Code42OnPollConnector, self).__init__()
```

Comment:
Does `Code42OnPollConnector` have a base class defined? If not, what is the purpose of calling `super(...).__init__()` here?

Reply:
Oops, that is no longer needed